codec/src/main/java/io/netty/handler/codec/Headers.java

```diff
@@ -19,7 +19,14 @@
 import java.util.Map.Entry;
 import java.util.Set;
 
-public interface Headers<T> extends Iterable<Entry<T, T>> {
+/**
+ * Common interface for {@link Headers} which represents a mapping of key to value.
+ * Duplicate keys may be allowed by implementations.
+ *
+ * @param <K> the type of the header name.
+ * @param <V> the type of the header value.
+ */
+public interface Headers<K, V> extends Iterable<Entry<K, V>> {
```

Comment:
Do we currently have a use case for anything other than `Headers<CharSequence, CharSequence>`? If not, perhaps we should drop one or both parameters altogether?